File: backend/src/main/java/com/xuma/pos/report/controller/ReportController.java
package com.xuma.pos.report.controller;

import com.xuma.pos.common.api.ApiResponse;
import com.xuma.pos.report.dto.DailySalesResponse;
import com.xuma.pos.report.dto.TopItemResponse;
import com.xuma.pos.report.service.ReportService;
import lombok.RequiredArgsConstructor;
import org.springframework.format.annotation.DateTimeFormat;
import org.springframework.http.ResponseEntity;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.time.LocalDateTime;
import java.util.List;

@RestController
@RequestMapping("/api/reports")
@RequiredArgsConstructor
public class ReportController {
    private final ReportService reportService;

    @GetMapping("/daily-sales")
    @PreAuthorize("hasAnyRole('MANAGER', 'ADMIN')")
    public ResponseEntity<ApiResponse<List<DailySalesResponse>>> getDailySales(
            @RequestParam @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime startDate,
            @RequestParam @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime endDate) {
        return ResponseEntity.ok(ApiResponse.ok(reportService.getDailySales(startDate, endDate)));
    }

    @GetMapping("/top-items")
    @PreAuthorize("hasAnyRole('MANAGER', 'ADMIN')")
    public ResponseEntity<ApiResponse<List<TopItemResponse>>> getTopSellingItems(
            @RequestParam(defaultValue = "10") int limit) {
        return ResponseEntity.ok(ApiResponse.ok(reportService.getTopSellingItems(limit)));
    }
}

File: backend/src/report/report.controller.ts
import { Controller, Get, Query, UseGuards } from '@nestjs/common';
import { ReportService } from './report.service';
import { JwtAuthGuard } from '../auth/guards/jwt-auth.guard';
import { RolesGuard } from '../auth/guards/roles.guard';
import { Roles } from '../auth/decorators/roles.decorator';

@Controller('api/reports')
@UseGuards(JwtAuthGuard, RolesGuard)
@Roles('ADMIN', 'MANAGER')
export class ReportController {
  constructor(private readonly reportService: ReportService) {}

  @Get('daily-sales')
  getDailySales(
    @Query('startDate') startDate: string,
    @Query('endDate') endDate: string
  ) {
    return this.reportService.getDailySales(startDate, endDate);
  }

  @Get('top-items')
  getTopSellingItems(@Query('limit') limit: string) {
    return this.reportService.getTopSellingItems(limit ? parseInt(limit, 10) : 10);
  }

  @Get('order-types')
  getOrderTypeAnalytics(
    @Query('startDate') startDate: string,
    @Query('endDate') endDate: string
  ) {
    return this.reportService.getOrderTypeAnalytics(startDate, endDate);
  }

  @Get('hourly-revenue')
  getHourlyRevenue(
    @Query('startDate') startDate: string,
    @Query('endDate') endDate: string
  ) {
    return this.reportService.getHourlyRevenue(startDate, endDate);
  }

  @Get('customers/ltv')
  getTopCustomersLtv(@Query('limit') limit: string) {
    return this.reportService.getTopCustomersLtv(limit ? parseInt(limit, 10) : 10);
  }

  @Get('customers/new-vs-returning')
  getNewVsReturningCustomers(
    @Query('startDate') startDate: string,
    @Query('endDate') endDate: string
  ) {
    return this.reportService.getNewVsReturningCustomers(startDate, endDate);
  }
}
